site/pages/api/change-password.api.ts

```diff
@@ -0,0 +1,42 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { COOKIES_CHANGE_PASSWORD_ERRORS, CHANGE_PASSWORD_PAGE_PATH } from "../../constants";
+import { changePasswordSchemaRefined } from "../../schemas/change-password.schema";
+import { flattenZodErrors } from "../../utils";
+import {
+    redirectToError,
+    setCookieOnResponseObject,
+    redirectTo,
+    destroyCookieOnResponseObject,
+} from "../../utils/apiUtils";
+
+const changePassword = (req: NextApiRequest, res: NextApiResponse) => {
+    try {
+        const validatedBody = changePasswordSchemaRefined.safeParse(req.body);
+        if (!validatedBody.success) {
+            setCookieOnResponseObject(
+                COOKIES_CHANGE_PASSWORD_ERRORS,
+                JSON.stringify({
+                    inputs: req.body as object,
```

Comment:
Do we not want to do any validation on the req.body before we put it in the cookie? If the user edits the form and sends across odd info, we will just put it in the cookie and later on when we try parse it, it will presumably fail?

Reply:
The cookie is set after validation and parsing. The cookie set is an error cookie where the inputs are what is passed back to the user

Reply:
The cookie set uses req.body, which the user sends to the API.

If we are just putting the req.body directly into the cookie, then if the user sends this :

{ 
dinner: "fish"
}

Then we will add it to the cookie, which isnt what we want (presumably it will break when we try and parse the cookie later and dont find what we expect).

Reply:
The same schema is used to parse later too - so if using this schema to parse would cause it to break then it would break in the api too before setting the cookie

Reply:
Noticed that the refined schema was not used which is now updated

Reply:
Right ok - cool! Looks good then